tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -0,0 +1,263 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Nuke.Common;
+using Nuke.Common.CI.AzurePipelines;
+using Nuke.Common.Execution;
+using Nuke.Common.Tools.Git;
+using NukeExtensions;
+
+partial class Build : NukeBuild
+{
+
+    [Parameter("Indicates matrices target. Match the name of generated variable", List = false)]
+    readonly GenerateMatricesTarget? GenerateMatricesTarget;
+
+    [Parameter("Indicates condition. Must be in the format '{variableName}|[filter1,filter2];{variableName}|[filter1,filter2]'", List = false)]
+    readonly string GenerateConditionVariableFilter;
+
+    Target GenerateMatrices
+        => _ =>
+        {
+            return _
+                  .Unlisted()
+                  .Requires(() => GenerateMatricesTarget)
+                  .Executes(() =>
+                  {
+                      if (!CheckChanges())
+                      {
+                          Logger.Info($"No changes found.");
+                          return;
+                      }
+
+                      var target = GenerateMatricesTarget ?? global::GenerateMatricesTarget.none;
+                      if (global::GenerateMatricesTarget.integration_tests_windows_matrices.HasFlag(target))
+                      {
+                          GenerateIntegrationTestsWindowsMatrices(target);
+                      }
+
+                      if (target.HasFlag(global::GenerateMatricesTarget.integration_tests_linux_matrix))
+                      {
+                          GenerateIntegrationTestsLinuxMatrix();
+                      }
+
+                      if (global::GenerateMatricesTarget.exploration_tests_matrices.HasFlag(target))
+                      {
+                          GenerateExplorationTestMatrices(target);
+                      }
+                  });
+
+            bool CheckChanges()
+            {
+                if (string.IsNullOrWhiteSpace(GenerateConditionVariableFilter))
+                {
+                    Logger.Info("Filter condition is not provided, assuming true.");
+                    return true;
+                }
+
+                var varFilterConditions = GenerateConditionVariableFilter.Split(';');
+
+                var isChanged = false;
+
+                foreach (var varFilterCondition in varFilterConditions)
+                {
+                    Logger.Info($"Checking changes for {varFilterCondition}");
+                    isChanged = CheckChange(varFilterCondition) || isChanged;
+                }
+
+                return isChanged;
```

Comment:
To build on @andrewlock's point on simplifying this, the way we are intermixing these conditions between the yaml and NUKE seems overly complicated. I suggest removing any mention of conditions from this file so that we always generate *all* the matrices, and only check the conditions in the yaml. 

As for the logic for evaluating `isProfilerChanged` and `isDebuggerChanged`, I propose we either revert back to the original yaml/powershell implementation, or if we want to keep it in C#, move it to a different file, as these conditions aren't strictly related to generating matrices and we will likely want to rely on them in other situations in the future.

Reply:
If I got @andrewlock right, he wanted to use conditions for other tests/tasks as well.
This code allow to generate those conditions.